mobile/lib/src/infrastructure/connection_repositories/eeg_data_repository.dart

```diff
@@ -26,26 +29,23 @@ class EEGDataRepository implements IEEGDataRepository {
   BaseOpenBCITransformer<List<int>, List<dynamic>> currentStreamTransformer;
   BaseOpenBCITransformer<List<int>, List<dynamic>> _currentTransformer;
   StreamSubscription _currentTransformerStream;
-  StreamingSharedPreferences _preferences;
   EEGData _recordingData;
   double _fpzCzChannelMax = 0;
   double _pzOzChannelMax = 0;
   int _dataCount = 0;
 
   @override
-  void initialize() async {
-    _preferences ??= await StreamingSharedPreferences.instance;
-
-    _currentTransformer =
-        _preferences.getBool('using_bluetooth', defaultValue: false).getValue()
-            ? _ganglionTransformer
-            : _cytonTransformer;
+  void initialize(DeviceType deviceType) async {
```

Comment:
Could we get rid of initialize and instead pass a deviceType to createRecordingFromStream? This would simplify how to use this repository.